imagetagger/imagetagger/images/management/commands/runzipdaemon.py

```diff
@@ -37,12 +38,14 @@ def _regenerate_zip(self, imageset):
             self.stderr.write('skipping regeneration of ready imageset {}'.format(imageset.name))
             return
 
-        with ZipFile(os.path.join(settings.IMAGE_PATH, imageset.tmp_zip_path()), 'w') as f:
-            for image in imageset.images.all():
-                f.write(image.path(), image.name)
-
-        os.rename(os.path.join(settings.IMAGE_PATH, imageset.tmp_zip_path()),
-                  os.path.join(settings.IMAGE_PATH, imageset.zip_path()))
+        tmp_zip_path = imageset.tmp_zip_path()
+        with tmp().open(tmp_zip_path, 'wb') as zip_file:
```

Comment:
Thanks for testing and catching this bug. I did in fact not test the zip daemon, because I'm not using this feature in my deployment. Unit tests would have helped as well. I saw the project didn't have any yet and I was a bit lazy to add a basic setup.

Reply:
No problem. I agree that we should introduce unit-tests an it's definitely on my list